### ISSUE TITLE
Split user search fields in Admin [20,13,1.54]

### ISSUE THREAD

**Issue by hellafitz** [Iridescent-CM/technovation-app#2620]
from epic #2599

- split the Name or Email fields apart into First Name and Last Name (searching results from any user type unless admin selects something specifically), Email
- split off the Parent Name (combined first and last)
- fields should search using exact match (which we'll confirm w/ PT and CET and Ops feedback after)

**Comment by stenington** [comment]
In implementing this, I noticed that changing this for admins will change it for chapter ambassadors too.

**Comment by stenington** [comment]
@hellafitz @rsgonzal I realized while working on this that the same code backs the chapter ambassador and admin participants pages. Would you want this change to happen for both types of user?

**Comment by stenington** [comment]
There is a way for me to do this only for admins, which is what I will default to for now since that's how the ticket is written. But if chapter ambassadors should also get these new search fields, that's pretty straight forward to do I think.

**Comment by rsgonzal** [comment]
I don't see anything wrong with chapter ambassadors getting the same changes. I think it would probably be helpful for them as well?

**Comment by stenington** [comment]
## QA steps:

1. log in as an admin
2. make sure you can use each of the new search fields, and get the results you would expect
3. log in as a chapter ambassador
4. make sure the search fields work the same over there

**Comment by kate-technovation** [comment]
Wow--thank you so much, Team!! I really, really appreciate this upgrade.

1. Done
2. Tried searching in First Name field for "North". Found exact results:
<img width="780" alt="Screen Shot 2020-09-08 at 6 59 45 PM" src="https://user-images.githubusercontent.com/57409606/92535555-79c60400-f205-11ea-9f62-acf3cccc318c.png">

Tried searching in Last Name field for "Carlsberg". Found exact results:
<img width="653" alt="Screen Shot 2020-09-08 at 7 00 26 PM" src="https://user-images.githubusercontent.com/57409606/92535591-906c5b00-f205-11ea-9817-65448fa0802c.png">

Tried searching in Email field for "ophelie+ca1@technovation.org". Found exact results:
<img width="630" alt="Screen Shot 2020-09-08 at 7 01 56 PM" src="https://user-images.githubusercontent.com/57409606/92535677-c578ad80-f205-11ea-816c-1cabdfafa530.png">

Added Parent or Guardian Name filter and searched on parent first name "Mom" and then parent last name "Tea" in the Parent or Guardian Name field. Found exact results both times:
<img width="674" alt="Screen Shot 2020-09-08 at 7 05 57 PM" src="https://user-images.githubusercontent.com/57409606/92535906-55b6f280-f206-11ea-9afe-ced82226ba6b.png">

<img width="662" alt="Screen Shot 2020-09-08 at 7 06 28 PM" src="https://user-images.githubusercontent.com/57409606/92535938-66ffff00-f206-11ea-9189-58c63f39a5e1.png">

I noticed that selecting the  Parent or Guardian Name filter also brings up the Parent or Guardian Email field. So WITHOUT selecting the Parent or Guardian Email filter, I searched on "rebecca+bobamom@technovation.org" in the Parent or Guardian Email field. Found exact results:
<img width="472" alt="Screen Shot 2020-09-08 at 7 08 33 PM" src="https://user-images.githubusercontent.com/57409606/92536054-b21a1200-f206-11ea-9339-213f871d0fcd.png">

And then, to be safe, I removed the  Parent or Guardian Name filter and added the  Parent or Guardian Email filter, and searched on "rebecca+bobamom@technovation.org" in the Parent or Guardian Email field. Found exact results:
<img width="815" alt="Screen Shot 2020-09-08 at 7 09 43 PM" src="https://user-images.githubusercontent.com/57409606/92536121-dbd33900-f206-11ea-9a78-0258a6a982cc.png">

3. Can't log in as a CA, even with approved background check--says it's still processing. I'll try again tomorrow (Wednesday).

**Comment by kate-technovation** [comment]
Hmm...logged in again today (Thursday), and still am not able to search as a CA, due to processing background check:

**Comment by kate-technovation** [comment]
Too funny--it just got sent. 

For Step 3: 
Logged in as a CA and had to edit and save program info to see dashboard:
<img width="784" alt="Screen Shot 2020-09-10 at 4 21 50 PM" src="https://user-images.githubusercontent.com/57409606/92799382-bcace680-f381-11ea-9356-fb9468469f77.png">

Clicked on View All Participants in your area and tried searching in First Name field for "North". Found zero results. 
<img width="641" alt="Screen Shot 2020-09-10 at 4 37 00 PM" src="https://user-images.githubusercontent.com/57409606/92801869-00085480-f384-11ea-93a5-b28bc4bcf1f3.png">

Noticed that "City" had been auto-added as a filter, so I removed it, and tried searching again. Found zero results AND the "City" filter had been auto-added back:
<img width="668" alt="Screen Shot 2020-09-10 at 4 38 36 PM" src="https://user-images.githubusercontent.com/57409606/92801954-144c5180-f384-11ea-894d-18ddd958da8a.png">

Tried searching in Last Name field for "Carlsberg". Found exact results:
<img width="651" alt="Screen Shot 2020-09-10 at 4 39 12 PM" src="https://user-images.githubusercontent.com/57409606/92802044-29c17b80-f384-11ea-8a9b-eff95a41fb85.png">

Tried searching in Email field for "ophelie+ca1@technovation.org". Found zero results:
<img width="647" alt="Screen Shot 2020-09-10 at 4 39 39 PM" src="https://user-images.githubusercontent.com/57409606/92802120-3940c480-f384-11ea-8fde-3a37457b4a50.png">

Noticed a tiny box that said Find a Missing Participant near bottom of screen:
<img width="667" alt="Screen Shot 2020-09-10 at 4 40 37 PM" src="https://user-images.githubusercontent.com/57409606/92802333-6f7e4400-f384-11ea-817d-7be019838fcb.png">

Clicked on it, entered "ophelie+ca1@technovation.org", and hit Search button:
<img width="566" alt="Screen Shot 2020-09-10 at 4 40 18 PM" src="https://user-images.githubusercontent.com/57409606/92802357-73aa6180-f384-11ea-9f30-15ee4b91d0c0.png">

Found correct result:
<img width="482" alt="Screen Shot 2020-09-10 at 4 40 31 PM" src="https://user-images.githubusercontent.com/57409606/92802370-76a55200-f384-11ea-98a7-63885f089156.png">

Navigated back to Search screen and added Parent or Guardian Name filter and searched on parent first name "Mom" in the Parent or Guardian Name field. Found a result for Mom, but NOT the same one as I did as an Admin: 
<img width="656" alt="Screen Shot 2020-09-10 at 4 42 26 PM" src="https://user-images.githubusercontent.com/57409606/92802565-9dfc1f00-f384-11ea-83d8-605df78699a2.png">

Tried the missing participant button, but since this is a parent and not a participant, the search could not find the other parent named "Mom" (probably not a big deal, since CAs don't need to find parents outside of their region, but just letting you know):
<img width="445" alt="Screen Shot 2020-09-10 at 4 44 23 PM" src="https://user-images.githubusercontent.com/57409606/92802870-e3205100-f384-11ea-9390-141057adaf0d.png">

Navigated back to Search screen and searched on parent last name "Tea" in the Parent or Guardian Name field. No results:
<img width="645" alt="Screen Shot 2020-09-10 at 4 45 25 PM" src="https://user-images.githubusercontent.com/57409606/92803130-1d89ee00-f385-11ea-9f5e-1c28a7e69e1e.png">

I noticed that selecting the Parent or Guardian Name filter also brings up the Parent or Guardian Email field. So WITHOUT selecting the Parent or Guardian Email filter, I searched on "rebecca+bobamom@technovation.org" in the Parent or Guardian Email field. No results:
<img width="647" alt="Screen Shot 2020-09-10 at 4 46 26 PM" src="https://user-images.githubusercontent.com/57409606/92803270-427e6100-f385-11ea-8c30-f5a62ef5e366.png">

And then, to be safe, I removed the Parent or Guardian Name filter and added the Parent or Guardian Email filter, and searched on "rebecca+bobamom@technovation.org" in the Parent or Guardian Email field. No results:
<img width="646" alt="Screen Shot 2020-09-10 at 4 46 48 PM" src="https://user-images.githubusercontent.com/57409606/92803296-490cd880-f385-11ea-9032-4345dc02b353.png">


TL;DR I thought that the lack of search results was due to me having set my CA in Montecito, CA, but when I went to change my city, I noticed that I technically have lots of participants in my region:

<img width="628" alt="Screen Shot 2020-09-10 at 4 49 23 PM" src="https://user-images.githubusercontent.com/57409606/92803644-a012ad80-f385-11ea-8d8f-9ee0b70b446a.png">
<img width="653" alt="Screen Shot 2020-09-10 at 4 49 29 PM" src="https://user-images.githubusercontent.com/57409606/92803647-a0ab4400-f385-11ea-9ef8-9f2bef08b29b.png">

**Comment by kate-technovation** [comment]
One final comment--the internal Admin side works great! The CA Admin side appears to work too, but since a lot of our test accounts have shared names/addresses, it was tough for me to see how specific the search will be for CAs (i.e. if several students have parents with the same name).

**Comment by ophelieh** [comment]
Hi all! I don't know that we should have exact results only for the names & that first name and last name should be separated. Sometimes we search someone and don't have a full name and might not know if it's a first name or last name. Then we have no way of finding that person (in the case we don't have an email). And it happens enough for me to be concerned about it. I'm sure it happens to Chapter Ambassadors as well. 

For example, I know we have someone of the first name Mentor, but when I search "Ment" I find 0 results
![Screen Shot 2020-09-11 at 8 12 15 AM (2)](https://user-images.githubusercontent.com/32206421/92929678-8b2a2e80-f406-11ea-822b-c8bcef2c88a0.png)
![Screen Shot 2020-09-11 at 8 12 41 AM (2)](https://user-images.githubusercontent.com/32206421/92929711-9715f080-f406-11ea-9287-065fef03c80f.png)

 --- how would I ever find that person? Or if I don't know how to spell someone's name? I understand the need for exact results especially when dealing with Zendesk, but I am concerned that it will cause more issues than it will solve for others. For example, what happens when someone sends in a form but you have to guess the name and the email doesn't match any on record? Yes, you can do back and forth but that's assuming the person will respond. If you only have exact results, you might never find that person based on close spelling... 

For email, the search is fine as if I just type the first portion of the email, I can see all emails with that. 
![Screen Shot 2020-09-11 at 8 14 17 AM (2)](https://user-images.githubusercontent.com/32206421/92929840-d17f8d80-f406-11ea-970b-7ecee018ab7b.png)

**Comment by ksrebecca** [comment]
As per Ophelie's comment is this an all or nothing sort of thing? How would we deal with non uniform spellings? (since so many people spell their own names differently when moving them to English...)

**Comment by kate-technovation** [comment]
While I agree with Ophélie's question regarding what happens when you type part of a name, I would like to highlight that exact search results cut down on a **lot** of time spent searching for end users. 

If there's a way for us to add specificity and be able remove it with search operators (i.e. "EXACT:") that's fine, but we do need the ability to drill down to exact search results on an everyday basis for answering the majority of our tickets (about 3,883 last season). 

Regarding Rebecca's point, yes--the way that some names (especially with accents on the letters or non-Roman letters) appear can sometimes affect search results (i.e. in Zendesk, we tend to receive the person's native letters for their name, and searching on those don't always return results. However, I understand if this is too big to address in this upgrade).

**Comment by rsgonzal** [comment]
thank you all for your feedback! we did talk about the possibility of having a fuzzy search as well, so that is something we could look into. As long as the Chapter Ambassador view works ok, I know it's hard to test with little data, then we will be moving this forward.

Our plan for next steps was to have @kate-technovation try some of her searches on production and test if it yields better results than before.

**Comment by kate-technovation** [comment]
Thanks, Roxana! Much appreciated.

Should I test in Production today, or later (i.e. do you need time to update it)?

**Comment by nickiredescent** [comment]
Just chiming in to also say that having only exact search will give us a difficult time when searching for the many variations of names and emails.

Kate already mentioned special characters. I created another dummy account, first name "Ophélie" and I cannot find her by searching "Ophelie"/without having the accented é in the name.

Email search seems to only yield multiple results if I give enough of the beginning of the email address. For example:
- Searching for "roxana" yields 20 results
- Searching for "roxana+" yields 20 results
- Searching for "roxan" yields 0 results
- Searching for "technovation" yields 0 results
- Searching for "@technovation" yields 0 results
- Searching for "technovation.org" yields 5 results (obviously too low)
- Searching for ".org" yields 0 results

In the case of separate first and last name fields, I especially think about names from Latin America where it's unclear how the user would input their information. How would the admin search for the name "Pablo Diego Jose Francisco" for example? While possible, it would be time consuming.

On a related note, the dummy account "[Just A] [Chapter Ambassador]" is difficult to search for via only first name. 
- Searching for "Just" yields 0 results
- Searching for "A" yields 0 results
- Searching for "Just A" yields the appropriate result
The account "[Neil Degrasse] [Tyson1]" does not have this issue.

**Comment by kate-technovation** [comment]
Just want to chime in on the first and last name fields point in Nick's note above--while it may seem confusing as to how we would search for "Pablo Diego Jose Francisco", I'd like to note that the opposite is also as true--that when we receive clear first names or last names from students and volunteers, typing them in a single "Name" field yields way too many results.

The goal of the separate name fields is to avoid this confusion on every search. While we do receive some longer form names, when those users created their account, they were asked to enter a separate first and last name. So our search would ideally match that circumstance.

**Comment by stenington** [comment]
Seems like there's two basic use cases here:
1. we know very little about the person we're looking up, and want to go broad with the search results, and...
2. we know a bit more about the person we're looking up, and want to go narrow and pull them up with out a lot of extra noise.

So maybe to move this forward, instead of replacing the "Name or Email" field, we have both the old "Name or Email" field **and** we have all the new, more specific exact search fields?

**Comment by kate-technovation** [comment]
Hi Mike--

Having both the old and new fields sounds like a great idea! Thank you:)

**Comment by ophelieh** [comment]
Love that idea @stenington!

**Comment by rsgonzal** [comment]
@kate-technovation @ophelieh @nickiredescent looks like both options are up, can you take one last quick look? We'll have some more asks from you once it's on production :)

**Comment by kate-technovation** [comment]
Will test first thing tomorrow:) Thanks, Roxana!!

**Comment by kate-technovation** [comment]
Thanks @rsgonzal !

Okay, so I'm first testing this as a Chapter Admin. I tried searching on part of your email address for one of our accounts "roxana+newca". When I input it into the Email only field, I got zero results:
<img width="799" alt="Screen Shot 2020-09-15 at 12 10 31 PM" src="https://user-images.githubusercontent.com/57409606/93236139-74f4d900-f74c-11ea-9ee0-bd5e5516cd6d.png">

However, when I searched on it in the Name or Email field, I got results, but not exact, even though you have only one account with this email address:
<img width="795" alt="Screen Shot 2020-09-15 at 12 11 04 PM" src="https://user-images.githubusercontent.com/57409606/93236213-88a03f80-f74c-11ea-8d69-d86a890d21ee.png">

I thought it might be the partial address throwing things off, so I tried searching with the whole email address in Name or Email, but still got too many results:
<img width="790" alt="Screen Shot 2020-09-15 at 12 11 32 PM" src="https://user-images.githubusercontent.com/57409606/93236263-99e94c00-f74c-11ea-9f0b-6f9a38200645.png">

I then searched in the Email only field with the whole address, and got the exact result for which I was looking:
<img width="788" alt="Screen Shot 2020-09-15 at 12 12 32 PM" src="https://user-images.githubusercontent.com/57409606/93236362-bdac9200-f74c-11ea-9b9e-25301bc4bab8.png">

So it's good to know--if I have a partial address, I can search in the Name or email field, but if I have an exact address, I can search in the Email only field.

I next logged in as an Admin and searched on "Stinky" in the Name or email field and got an exact result:
<img width="780" alt="Screen Shot 2020-09-15 at 12 13 58 PM" src="https://user-images.githubusercontent.com/57409606/93236526-f0ef2100-f74c-11ea-8b3d-25d2f1026e99.png">

Same thing when I searched on "Stinky" in the First name only field:
<img width="791" alt="Screen Shot 2020-09-15 at 12 14 28 PM" src="https://user-images.githubusercontent.com/57409606/93236587-03695a80-f74d-11ea-81b3-ea9e268df841.png">

I searched on rebecca+6 in the Name or email field but got inexact results:
![Uploading Screen Shot 2020-09-15 at 12.15.21 PM.png…]()

Next, I searched on rebecca+6 in the Email only field and got exact results:
<img width="783" alt="Screen Shot 2020-09-15 at 12 36 37 PM" src="https://user-images.githubusercontent.com/57409606/93238942-1d586c80-f750-11ea-93fe-fef14350f2c3.png">

This is interesting, because it's different than what happened when I was logged in as a CA (i.e. searching for a partial email address in the Email only field).

I then searched on the full email address "rebecca+6@technovation.org" in the Name or email field, and got too many results:
<img width="803" alt="Screen Shot 2020-09-15 at 12 38 02 PM" src="https://user-images.githubusercontent.com/57409606/93239085-4da00b00-f750-11ea-811c-0cc12b2e58c1.png">

Lastly, I searched on the full email address "rebecca+6@technovation.org" in the Email only field and got exact results:
<img width="780" alt="Screen Shot 2020-09-15 at 12 38 32 PM" src="https://user-images.githubusercontent.com/57409606/93239128-5ee91780-f750-11ea-8d46-53f5b7141d65.png">

**Comment by rsgonzal** [comment]
Thanks, kate! this is good stuff. Definitely good to keep note of which field is good for different types of searches

**Comment by stenington** [comment]
Parking some more info here for later. This is how I think the "Name or Email" field is currently working:

![Screen Shot 2020-09-15 at 1 33 06 PM](https://user-images.githubusercontent.com/55319/93244706-277e6900-f758-11ea-953c-30eba22fda60.png)

The black text is examples of searches you might type. The colored underscores show which word(s) get checked against which field(s) as a fuzzy match.
